sagemaker-inference-recommender/tensorflow-cloudwatch/cloudwatch.py

```diff
@@ -0,0 +1,314 @@
+            env_parameters.append(endpoint_configuration['ModelConfiguration']
+                                  ['EnvironmentParameters'])
+
+        if len(endpoint_names) == 0:
+            print(f'No endpoint recommendations for {job_name}')
+            exit(-1)
+
+        data = {
+            "InstanceType": instance_type,
+            "MaximumInvocations": max_invocations,
+            "ModelLatency": model_latency,
+            "CostPerHour": cost_per_hour,
+            "CostPerInference": cost_per_inference,
+            "EndpointName": endpoint_names,
+            "VariantName": variant_names,
+            "InitialCount": initial_count,
+#             "isCompiled": is_compiled,
+            "EnvParameters": env_parameters,
+            "StartTime": start_time,
+            "EndTime": end_time
+        }
+        df = pd.DataFrame(data)
+
+        # print(f'Result in dataframe {df}')
+        return df
+    else:
+        print(f"Job {job_name} is in {job_details['Status']} status, we can plot only completed jobs")
+        exit(0)
+
+
+def get_cw_metrics(cw_client, endpoint_name, variant_name, metrics_name, start_time, end_time):
+    if endpoint_metrics[metrics_name]['Statistics'] != 'None':
+        cw_data = cw_client.get_metric_statistics(
+            Namespace=endpoint_metrics[metrics_name]['Namespace'],
+            MetricName=metrics_name,
+            Dimensions=[
+                {
+                    'Name': 'EndpointName',
+                    'Value': endpoint_name
+                },
+                {
+                    'Name': 'VariantName',
+                    'Value': variant_name
+                }
+            ],
+            StartTime=start_time,
+            EndTime=end_time,
+            Period=endpoint_metrics[metrics_name]['Period'],
+            Statistics=[
+                endpoint_metrics[metrics_name]['Statistics'],
+            ],
+            Unit=endpoint_metrics[metrics_name]['Unit']
+        )
+    elif endpoint_metrics[metrics_name]['ExtendedStatistics'] != 'None':
+        cw_data = cw_client.get_metric_statistics(
+            Namespace=endpoint_metrics[metrics_name]['Namespace'],
+            MetricName=metrics_name,
+            Dimensions=[
+                {
+                    'Name': 'EndpointName',
+                    'Value': endpoint_name
+                },
+                {
+                    'Name': 'VariantName',
+                    'Value': variant_name
+                }
+            ],
+            StartTime=start_time,
+            EndTime=end_time,
+            Period=endpoint_metrics[metrics_name]['Period'],
+            ExtendedStatistics=[
+                endpoint_metrics[metrics_name]['ExtendedStatistics'],
+            ],
+            Unit=endpoint_metrics[metrics_name]['Unit']
+        )
+    else:
+        print(f'Both ExtendedStatistics & Statistics are None')
+        exit(0)
+
+    # print(f'Result of get_cw_metrics {cw_data}')
```

Comment:
Nit: delete the few print statements and other lines which are commented out within this file?

Reply:
thanks, cleaned up now